:bug: Fix to set correct UID and resource version when syncing updated objects and do not downsync upsynced objects

yana1205 opened the pull request:
<!--
Thanks for creating a pull request!

If this is your first time, please make sure to review CONTRIBUTING.MD.

Please copy the appropriate `:text:` or icon to the beginning of your PR title:

:sparkles: ✨ feature
:bug: 🐛 bug fix
:book: 📖 docs
:memo: 📝 proposal
:warning: ⚠️ breaking change
:seedling: 🌱 other/misc
:question: ❓ requires manual review/categorization

-->
## Summary

As reported in #625, there is a bug that upsynced objects are not updated. I fixed the bug and made sure that it addressed the issue with the issue opener (Tayebeh).   

What 's the changes
- Fix to set UID and Resource Version of the target k8s cluster's one when downsyncer/upsyncer push updated objects from upstream/downstream 
- Fix not to downsync upsynced objects (exactly saying not downsync objects having `edge.kcp.io/upsynced` annotation) and similarly not to upsync downsynced objects (`edge.kcp.io/downsynced` annotation) 
- Update tests

## Related issue(s)

Fixes #625

MikeSpreitzer commented:
The UID and ResourceVersion of an object are things that the apiserver writes, I am surprised to see words here about setting them.

yana1205 commented:
Yes, UID and Resource Version are determined at a server side. However, the UID and Resource Version of an object are different between upstream (mbws) and downstream (workload execution cluster). That's why I fixed the code so that when upsyncer syncs a downstream object to an upstream object, upsyncer replaces the UID and Resource Version of the downstream object with those of the upstream object and then pushes it to the upstream.

I have rewritten the title and description.

clubanderson commented:
/retest

MikeSpreitzer commented:
@yana1205 : thanks for the clarification.  Now I understand and it makes sense.
/lgtm

MikeSpreitzer commented:
/approve
